src/inmanta/ast/type.py

```diff
@@ -474,6 +616,35 @@ def __eq__(self, other: object) -> bool:
             return NotImplemented
         return self.element_type == other.element_type
 
+    def is_primitive(self) -> bool:
+        return self.get_base_type().is_primitive()
+
+    def corresponds_to(self, pytype: type[object]) -> bool:
+        if typing_inspect.is_union_type(pytype):
+            return False
+
+        if not typing_inspect.is_generic_type(pytype):
+            return issubclass(pytype, Sequence)
+
+        origin = typing_inspect.get_origin(pytype)
+        if not issubclass(origin, Sequence):
+            return False
+
+        args = typing_inspect.get_args(pytype, True)
+        if len(args) != 1:
+            return False
+        return self.element_type.corresponds_to(args[0])
+
+    def as_python_type_string(self) -> "str | None":
+        return f"list[{self.element_type.as_python_type_string()}]"
+
+    def to_python(self, instance: object) -> "object":
+        assert isinstance(instance, Sequence), f"This method can only be called on iterables, not on {type(instance)}"
```

Comment:
Shouldn't these be `ValueError` instead of `assert`?

Reply:
Good question

I think either we consider this a pre-condition and the assertion could be considered appropriate.

I don't think value error is correct: 
```
exception ValueError

    Raised when an operation or function receives an argument that has the right type but an inappropriate value, and the situation is not described by a more precise exception such as [IndexError](https://docs.python.org/3/library/exceptions.html#IndexError).
```

But type error is probably most specific. I'll change it to that